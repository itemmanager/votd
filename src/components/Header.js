import {Nav} from "./Nav";
import {Link} from "react-router-dom";
import React from "react";
import styled from "styled-components";
import {useIsAuthenticated} from "../auth";
import {Logout} from "./Logout";
import {Login} from "./Login";

const StyledHeader = styled.header`
  ul:first-child {
    padding: 0;
  }

  img {
    height: 2em;
  }
`;

export function Header() {
    const isAuthenticated = useIsAuthenticated();
    return <StyledHeader>
        <Nav>
            <ul>
                <li>
                    <Link to="/">
                        <img src={require("../images/home-logo.png")} alt="votd logo"/>
                    </Link>
                </li>
                <li className="grow"/>
                <li>
                    <Link to="/names">
                        Glyph Names
                    </Link>
                </li>
                {isAuthenticated?<Logout />:<Login />}
            </ul>
        </Nav>
    </StyledHeader>
}
