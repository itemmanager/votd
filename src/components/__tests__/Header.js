import renderer from 'react-test-renderer';
import { Header } from "../Header";
import { MemoryRouter } from 'react-router-dom';
import 'jest-styled-components'


test('Header snapshot', () => {
    const component = renderer.create(
        <MemoryRouter>
            <Header />
        </MemoryRouter>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})
