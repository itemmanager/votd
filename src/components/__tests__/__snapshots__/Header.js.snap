// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
.c1 {
  background-color: black;
}

.c1 a {
  color: white;
  display: block;
}

.c1 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  list-style-type: none;
  margin: 0;
  padding: 0.5em;
  gap: 1em;
}

.c1 li {
  padding: 0.5em;
}

.c1 .grow {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  background-color: #0F1D17;
  border: 1px solid rgb(96,200,148);
  color: #f5f5f5;
  border-radius: 5px;
  padding: 0.5em 1em;
  cursor: pointer;
}

.c3:hover {
  background-color: rgba(95,201,147,.55);
}

.c2 button {
  margin-right: 1em;
}

.c0 ul:first-child {
  padding: 0;
}

.c0 img {
  height: 2em;
}

<header
  className="c0"
>
  <nav
    className="c1"
  >
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="votd logo"
            src="home-logo.png"
          />
        </a>
      </li>
      <li
        className="grow"
      />
      <li>
        <a
          href="/names"
          onClick={[Function]}
        >
          Glyph Names
        </a>
      </li>
      <div
        className="c2"
      >
        <button
          className="c3"
          onClick={[Function]}
        >
          Login
        </button>
      </div>
    </ul>
  </nav>
</header>
`;
